fontforge/stemdb.c

```diff
@@ -4616,14 +4616,14 @@ static void MarkDStemCorner( struct glyphdata *gd,struct pointdata *pd ) {
     for ( i=0; i<pd->prevcnt && !has_stem; i++ ) {
 	stem = pd->prevstems[i];
 	if ( !stem->toobig && (
-	    ( x_dir && ( hv = IsUnitHV( &stem->unit,true ) == 1 )) ||
+	    ( x_dir && ( (hv = IsUnitHV( &stem->unit,true )) == 1 )) ||
 	    ( !x_dir && hv == 2 )))
```

Comment:
The bracket fix looks fine, but looking at the statement as a whole, it seems a bit odd that `IsUnitHV` (and the assignment to `hv`) is only called after checking if `x_dir` is true.

I wonder if it should actually be something closer to
````c
if ((hv = IsUnitHV(&stem->unit, true)) == 1 && x_dir) || (!x_dir && hv == 2)
````

I haven't studied the code long enough to know if this is appropriate.